### ISSUE TITLE
[feat] support pass through custom remark config in eslint config

### ISSUE THREAD

**Issue by KevinGhadyani-minted** [mdx-js/eslint-mdx#333]
### Subject of the issue

The Remark config is only seen if it's local to the app running ESLint, not an ESLint config package that included `eslint-plugin-mdx`.

A possible solution is accepting a custom Remark config rather than letting eslint-mdx determine where the config exists.

### Your environment

*   **OS**: Windows 10 64-bit.
*   **Packages**: `"eslint-plugin-mdx": "^1.14.1"`.
*   **Env**: node `v14.17.4` and yarn `1.22.5`.

### Steps to reproduce

1. Add Remark config to ESLint package directory.
2. Mess up a markdown file.
3. No fixes.

### Expected behaviour

Remark should look in the directory where the ESLint config for `eslint-mdx` is located rather than the current running process.

### Actual behaviour

Remark's config is ignored unless it's placed in the ESLint package consumer's root directory.

**Comment by JounQin** [comment]
Please provide a minimal reproduction.

**Comment by KevinGhadyani-minted** [comment]
Can you add an option to pass in a custom Remark config rather than using the `cosmiconfig`?

I have an `eslint-config-A` package being imported in to `app-X`. I want the Remark config to be in `eslint-config-A`, not `app-X`.

`cosmic` config only looks in `app-X` for the `.remarkrc.js`, not in `eslint-config-A`.

I'm sharing `eslint-config-A`, and want strict control over the Remark config. In no way should apps be in charge of it. I don't want to copy-paste a Remark config in 50 other apps (not exaggerating).

Currently, I still have to copy the config, but my workaround is doing this in `app-X`:
```js
module.exports = require('eslint-config-A/.remarkrc.js')
```

**Comment by JounQin** [comment]
Considering `.remarkrc.js` is the standard way of `remark`, I don't think we should support that.

I'm always using something like

```jsonc
// .remarkrc, or you can put it into `package.json`
{
  "plugins": ["@1stg/remark-config"]
}
```

But if you want to raise a PR to support it, we can give it a try.

**Comment by KevinGhadyani-minted** [comment]
Ok, that makes sense. So there _is_ a way to get a custom Remark config. The way I'd do it is by creating a separate Remark library and importing it in to my `eslint-config-A` library. Makes sense.

Closing!

**Comment by JounQin** [comment]
> importing it in to my eslint-config-A library. Makes sense.

That won't work AFAIK, it's just similar like your current workaround, you need to add it in pwd.

**Comment by KevinGhadyani-minted** [comment]
Oh yeah, good point. Darn...
